packages/framework/src/Component/Router/FriendlyUrl/FriendlyUrlRouterFactory.php

```diff
@@ -48,10 +66,44 @@ public function createRouter(DomainConfig $domainConfig, RequestContext $context
         return new FriendlyUrlRouter(
             $context,
             $this->configLoader,
-            new FriendlyUrlGenerator($context, $this->friendlyUrlRepository),
+            new FriendlyUrlGenerator(
```

Comment:
Isn't possible to use generator as a service here?
I think it would be more clear and we can inject just the dependency we need for this class (right now I need to inject two dependencies just to pass it to another constructor)

Reply:
No it is not.

It would be nice but there is `$context` needed which is passed during factoring the `FriendlyUrlRouter`.